smac/utils/configspace.py

```diff
@@ -169,12 +169,16 @@ def print_config_changes(
     if incumbent is None or challenger is None:
         return
 
-    params = sorted([(param, incumbent[param], challenger[param]) for param in challenger.keys()])
-    for param in params:
-        if param[1] != param[2]:
-            logger.info("--- %s: %r -> %r" % param)
-        else:
-            logger.debug("--- %s Remains unchanged: %r", param[0], param[1])
+    all_keys = set(incumbent.keys().union(challenger.keys()))
```

Comment:
In reference to #1041, I would recommend a more efficient way for obtaining the unique keys from `incumbent` and `challenger`.  Instead of calling `.keys()` explicitly twice, the first set on both dictionaries using the sequential union, will automatically provide the unique keys:

```python
all_keys = set(incumbent).union(challenger)
```

In this expression, both `set(incumbent)` and `challenger` in the `union` method will use the keys of the respective dictionaries. This way, we avoid redundant calls to `.keys()` and make the code more readable. 

Moreover, while Python may optimize away the unnecessary `.keys()` calls (i.e, most probably in the compiler's component), removing them now can potentially improve efficiency and certainly clarifies the intention of the code 👌

Reply:
Yup might be a bit faster, I'll implement it in, thank you :)